srcpkgs/pinephone-uboot/template

```diff
@@ -0,0 +1,114 @@
+license="GPL-2.0-only, BSD-3-Clause"
+homepage="http://www.denx.de/wiki/U-Boot/WebHome"
+
+# FOR MAINTAINERS:
+#
+# This u-boot package relies on downstream support for the pinephone hardware
+# from PINE64 and the Crust maintainers. Until this support is upstreamed,
+# please get sources for any new versions from PINE64's u-boot repo and Crust's
+# branch of arm-trusted-firmware.
+
+_commit_uboot=f0b298c504daae9da171bac950c036432cc00201
+_commit_armtf=f928897fa5df6afa0f70bb1b1011e4bb03cb4596
+_commit_crust=e63106c70bc4ddfa3575e4f1f3667de0795252a8
+
+_uboot=u-boot-$_commit_uboot
+_armtf=arm-trusted-firmware-$_commit_armtf
+_crust=crust-$_commit_crust
+
+distfiles="https://gitlab.com/pine64-org/u-boot/-/archive/${_commit_uboot}/u-boot-${_commit_uboot}.tar.gz>${_uboot}.tar.gz
+					 https://github.com/crust-firmware/arm-trusted-firmware/archive/${_commit_armtf}.tar.gz>${_armtf}.tar.gz
+					 https://github.com/crust-firmware/crust/archive/${_commit_crust}.tar.gz>${_crust}.tar.gz"
+
+checksum="f62fa00d0eda435faa823781cdf24982e87a6c1be9fb42ead6f9b14433457a6c
+					5161a07ed9065c876cd0d37bd9b544b77a707bcde9d0cdf86b83c7b0b5bfa95b
+					6fc2d68725a78b1ca6e7c6602286fc2aa2ddcdea28e83a85f6b988fcc5a513b8"
+
+_or1k="CROSS_COMPILE=or1k-none-elf-"
+
+do_patch() {
+	patch -d $_armtf -N -p1 < $FILESDIR/0001-ATF-set-fno-stack-protector.patch
+
+	patch -d $_uboot -N -p1 < $FILESDIR/0001-configs-Add-Pinetab-configuration.patch
+	patch -d $_uboot -N -p1 < $FILESDIR/fix-atf-bug.patch
+	patch -d $_uboot -N -p1 < $FILESDIR/0001-sunxi-support-asymmetric-dual-rank-DRAM-on-A64-R40.patch
+}
+
+# Note about _cross/_hostcc
+#
+# The Allwinner A64 has an "embedded controller" of sorts for managing
+# peripherals when the A53 cores are turned off. This EC is a custom CPU core
+# that uses the OpenRISC 1000 ISA, and it's what the Crust firmware runs
+# on. or1k is the gcc platform for OpenRISC, and it has nothing to do with any
+# real Void system targets. This is why the two cross-compilers are needed, or1k
+# is used ONLY for the crust portion of the firmware.
+
+do_configure() {
+	unset CFLAGS CXXFLAGS CPPFLAGS LDFLAGS
+	local _cross _hostcc
+
+	if [ "$CROSS_BUILD" ]; then
+		_cross="CROSS_COMPILE=${XBPS_CROSS_TRIPLET}-"
+		_hostcc="HOST_COMPILE=${XBPS_CROSS_TRIPLET}-"
+	fi
+
+	make -C $_crust $_hostcc $_or1k pinephone_defconfig
+	make -C $_uboot $_cross pinephone_defconfig
+	echo 'CONFIG_IDENT_STRING=" Voidlinux"' >> $_uboot/.config
+}
+
+do_build() {
+	unset CFLAGS CXXFLAGS CPPFLAGS LDFLAGS
+	local _cross _hostcc
+
+	if [ "$CROSS_BUILD" ]; then
+		_cross="CROSS_COMPILE=${XBPS_CROSS_TRIPLET}-"
+		_hostcc="HOST_COMPILE=${XBPS_CROSS_TRIPLET}-"
+	fi
+
+	make -C $_armtf $_cross PLAT=sun50i_a64 DEBUG=1 -j $makejobs bl31
+	make -C $_crust $_hostcc $_or1k -j $makejobs scp
+
+	export BL31=$wrksrc/$_armtf/build/sun50i_a64/debug/bl31.bin
+	export SCP=$wrksrc/$_crust/build/scp/scp.bin
+
+	if [ "$XBPS_TARGET_LIBC" = "musl" ]; then
+		touch $_uboot/include/stddef.h
+	fi
+
```

Comment:
Waiting for https://github.com/void-linux/void-packages/pull/24307

Reply:
@sgn should I remove this and rebase when #24307 is merged?